packages/web-js-sdk/src/sdk/fedcm.ts

```diff
@@ -0,0 +1,152 @@
+import { JWTResponse, SdkResponse, LoginOptions } from '@descope/core-js-sdk';
+import { CoreSdk } from '../types';
+
+/**
+ * Configuration for OneTap.
+ */
+interface OneTapConfig {
+  /** Whether to auto select. Optional. */
+  auto_select?: boolean;
+
+  /** Whether to cancel on tap outside. Optional. */
+  cancel_on_tap_outside?: boolean;
+
+  /** ID of the prompt parent. Optional. */
+  prompt_parent_id?: string;
+
+  /** Context. Optional. */
+  context?: 'signin' | 'signup' | 'use';
+
+  /** Callback function to handle the intermediate iframe close event. Optional. */
+  intermediate_iframe_close_callback?: () => void;
+
+  /** Whether to support ITP. Optional. */
+  itp_support?: boolean;
+
+  /** Login hint. Optional. */
+  login_hint?: string;
+
+  /** HD. Optional. */
+  hd?: string;
+
+  /** Whether to use FedCM for prompt. Optional. */
+  use_fedcm_for_prompt?: boolean;
+}
+
+/**
+ * Response from the credential.
+ */
+interface CredentialResponse {
+  /** Credential. */
+  credential: string;
+
+  /** How the selection was made. */
+  select_by:
+    | 'auto'
+    | 'user'
+    | 'user_1tap'
+    | 'user_2tap'
+    | 'btn'
+    | 'btn_confirm'
+    | 'btn_add_session'
+    | 'btn_confirm_add_session';
+}
+
+type OneTapInitialize = ({
+  client_id,
+  callback,
+  nonce,
+}: {
+  client_id: string;
+  callback: (res: CredentialResponse) => void;
+  nonce: string;
+} & OneTapConfig) => void;
+
+type PromptNotification = {
+  isSkippedMoment: () => boolean;
+};
+
+/**
+ * Constructs a higher level FedCM API that wraps the functions from code-js-sdk.
+ * @param sdk The CoreSdk instance.
+ * @returns The FedCM API.
+ */
+const createFedCM = (sdk: CoreSdk) => ({
+  async oneTap(
+    provider: string,
```

Comment:
Is this the API we want?
passing the provider is a bit useless in this case no?
I guess we will have a dedicated provider for it

WDYT?

Reply:
Made it optional with default to google